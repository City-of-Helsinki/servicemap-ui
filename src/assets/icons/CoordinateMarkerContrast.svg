<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle cx="12" cy="12" r="11.5" fill="#313131" stroke="white"/>
<mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="7" y="5" width="10" height="14">
<path fill-rule="evenodd" clip-rule="evenodd" d="M12 5.33333C9.42004 5.33333 7.33337 7.42 7.33337 10C7.33337 13.5 12 18.6667 12 18.6667C12 18.6667 16.6667 13.5 16.6667 10C16.6667 7.42 14.58 5.33333 12 5.33333ZM12 11.6667C11.08 11.6667 10.3334 10.92 10.3334 10C10.3334 9.08 11.08 8.33333 12 8.33333C12.92 8.33333 13.6667 9.08 13.6667 10C13.6667 10.92 12.92 11.6667 12 11.6667Z" fill="white"/>
</mask>
<g mask="url(#mask0)">
<rect x="-4.66663" y="-4.66667" width="33.3333" height="33.3333" fill="white"/>
<mask id="mask1" mask-type="alpha" maskUnits="userSpaceOnUse" x="-5" y="-5" width="34" height="34">
<rect x="-4.66663" y="-4.66667" width="33.3333" height="33.3333" fill="white"/>
</mask>
<g mask="url(#mask1)">
</g>
</g>
</svg>
