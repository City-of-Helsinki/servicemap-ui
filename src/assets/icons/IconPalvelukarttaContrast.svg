<svg width="25" height="24" viewBox="0 0 25 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_146_107072)">
<path d="M16.2002 14.1136C15.9075 14.4415 15.6121 14.7671 15.3275 15.102C14.1625 16.4782 13.085 17.9111 12.2193 19.3958C11.3536 17.9106 10.2761 16.4782 9.11112 15.102C8.82707 14.7671 8.53163 14.4415 8.23835 14.1136H4.90967C5.71237 14.9558 6.50694 15.8166 7.25099 16.6942C8.91669 18.654 10.3293 20.6836 11.0842 22.5987C11.2694 23.0667 11.718 23.3722 12.2198 23.3722C12.7217 23.3722 13.1697 23.0667 13.3555 22.5987C14.1104 20.6836 15.523 18.6546 17.1887 16.6942C17.9327 15.8171 18.7273 14.9558 19.53 14.1136H16.2002Z" fill="black"/>
<path d="M17.9225 0C15.9211 0 14.1321 0.907637 12.9368 2.334C13.7259 3.54491 14.1886 4.98982 14.1886 6.54545C14.1886 8.10109 13.7259 9.546 12.9368 10.7569C14.1321 12.1833 15.9211 13.0909 17.9225 13.0909C21.5232 13.0909 24.4397 10.1607 24.4397 6.54545C24.4397 2.93018 21.5232 0 17.9225 0Z" fill="#A4A6AB"/>
<path d="M6.51724 13.0909C10.1166 13.0909 13.0345 10.1604 13.0345 6.54545C13.0345 2.9305 10.1166 0 6.51724 0C2.91787 0 0 2.9305 0 6.54545C0 10.1604 2.91787 13.0909 6.51724 13.0909Z" fill="#141823"/>
</g>
<defs>
<clipPath id="clip0_146_107072">
<rect width="24.4397" height="23.3722" fill="white"/>
</clipPath>
</defs>
</svg>
