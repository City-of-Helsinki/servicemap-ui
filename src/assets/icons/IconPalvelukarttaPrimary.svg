<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
  <circle cx="20" cy="20" r="20" fill="none"/>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M23.0149 21.789C22.7756 22.0563 22.5341 22.321 22.3014 22.5936C21.3489 23.7139 20.468 24.8804 19.7602 26.0895C19.0524 24.8799 18.1714 23.7139 17.219 22.5936C16.9867 22.321 16.7452 22.0563 16.5054 21.789H13.7839C14.4402 22.4746 15.0898 23.1753 15.6982 23.8897C17.06 25.4851 18.2149 27.1374 18.8321 28.6968C18.9836 29.0774 19.3503 29.3265 19.7606 29.3265C20.1709 29.3265 20.5372 29.0774 20.6891 28.6968C21.3063 27.1374 22.4612 25.4856 23.8231 23.8897C24.4314 23.1757 25.081 22.4746 25.7373 21.789H23.0149Z" fill="black"/>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M24.4229 10.2996C22.7866 10.2996 21.324 11.0385 20.3467 12.1996C20.9919 13.1854 21.3702 14.3616 21.3702 15.628C21.3702 16.8944 20.9919 18.0706 20.3467 19.0564C21.324 20.2175 22.7866 20.9564 24.4229 20.9564C27.3668 20.9564 29.7513 18.5711 29.7513 15.628C29.7513 12.685 27.3668 10.2996 24.4229 10.2996Z" fill="#F7C74B"/>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M15.0982 10.2996C18.0408 10.2996 20.4266 12.6854 20.4266 15.628C20.4266 18.5706 18.0408 20.9564 15.0982 20.9564C12.1556 20.9564 9.76978 18.5706 9.76978 15.628C9.76978 12.6854 12.1556 10.2996 15.0982 10.2996Z" fill="#0016B7"/>
</svg>
