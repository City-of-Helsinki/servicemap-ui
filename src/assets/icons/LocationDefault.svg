<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle cx="16" cy="16" r="15" fill="white" stroke="#2242C7" stroke-width="2"/>
<mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="12" y="8" width="8" height="16">
<path fill-rule="evenodd" clip-rule="evenodd" d="M18.8364 9.84851C18.0606 9.06062 17.1152 8.66669 16 8.66669C15.2727 8.66669 14.6 8.8485 13.9818 9.21214C13.3636 9.57578 12.8788 10.0667 12.5273 10.6849C12.1758 11.3031 12 11.9758 12 12.7031C12 13.2364 12.103 13.7515 12.3091 14.2485C12.5152 14.7455 12.8 15.1697 13.1636 15.5212C13.5273 15.8727 13.9576 16.1576 14.4545 16.3758C14.9515 16.594 15.4667 16.703 16 16.703C17.1152 16.703 18.0606 16.3091 18.8364 15.5212C19.6121 14.7333 20 13.7879 20 12.6849C20 11.5818 19.6121 10.6364 18.8364 9.84851ZM16 17.4667C15.7576 17.4667 15.503 17.4424 15.2364 17.394V22.6303C15.2364 22.8485 15.3091 23.0364 15.4545 23.194C15.6 23.3515 15.7818 23.4303 16 23.4303C16.2182 23.4303 16.4 23.3515 16.5455 23.194C16.6909 23.0364 16.7636 22.8485 16.7636 22.6303V17.394C16.497 17.4424 16.2424 17.4667 16 17.4667Z" fill="white"/>
</mask>
<g mask="url(#mask0)">
<rect x="8" y="8" width="16" height="16" fill="#2242C7"/>
</g>
</svg>
