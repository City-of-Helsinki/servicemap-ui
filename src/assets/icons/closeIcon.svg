<svg width="18" height="18" viewBox="0 0 18 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="1" y="1" width="16" height="16">
<path fill-rule="evenodd" clip-rule="evenodd" d="M9 1.5C4.8525 1.5 1.5 4.8525 1.5 9C1.5 13.1475 4.8525 16.5 9 16.5C13.1475 16.5 16.5 13.1475 16.5 9C16.5 4.8525 13.1475 1.5 9 1.5ZM12.75 11.6925L11.6925 12.75L9 10.0575L6.3075 12.75L5.25 11.6925L7.9425 9L5.25 6.3075L6.3075 5.25L9 7.9425L11.6925 5.25L12.75 6.3075L10.0575 9L12.75 11.6925Z" fill="white"/>
</mask>
<g mask="url(#mask0)">
<rect x="-9.75" y="-9.75" width="37.5" height="37.5" fill="white"/>
<mask id="mask1" mask-type="alpha" maskUnits="userSpaceOnUse" x="-10" y="-10" width="38" height="38">
<rect x="-9.75" y="-9.75" width="37.5" height="37.5" fill="white"/>
</mask>
<g mask="url(#mask1)">
</g>
</g>
</svg>
