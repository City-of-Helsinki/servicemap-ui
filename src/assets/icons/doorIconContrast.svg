<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle cx="12" cy="12" r="11.25" fill="white" stroke="#4C4D4F" stroke-width="1.5"/>
<path d="M9.80555 8.81862C10.6575 8.81862 11.3626 8.12182 11.3626 7.25819C11.3626 6.39455 10.6673 5.69775 9.80555 5.69775C8.9536 5.69775 8.24854 6.39455 8.24854 7.25819C8.24854 8.12182 8.9536 8.81862 9.80555 8.81862Z" fill="#4C4D4F"/>
<path d="M10 14.0055C9.8727 13.9467 9.52156 14.3303 9.46281 14.4481C9.37468 14.6346 9.0809 15.2823 8.84588 15.7043C8.54231 16.2539 7.21052 18.0302 6.92654 18.1578C6.83841 18.1872 7.4085 18.2461 7.30078 18.1578C7.21265 18.0793 7.49021 18.0743 7.5 18.0056C7.54896 17.829 8.14747 17.4669 8.5 17.0056C8.79378 16.6229 8.73816 16.2833 8.84588 16.0968C9.11028 15.6552 10.0034 14.0967 10.0328 14.0378C10.0916 13.9102 9.1273 13.0645 9 13.0056C8.8727 12.9467 8.63044 13.3685 8.57169 13.4961C8.56189 13.5059 7.97434 14.8308 7.71974 15.243C7.62181 15.4 7.3672 15.7337 7.08322 16.0968C6.46629 16.9212 6.07459 17.4414 6.01583 17.6867C5.95708 17.9321 6.0648 18.2167 6.29002 18.413C6.46629 18.57 6.67193 18.6485 6.86778 18.6485C6.94612 18.6485 7.02447 18.6387 7.10281 18.6092C7.58264 18.4326 9.03194 16.3716 9.27675 15.93C9.51177 15.4982 9.81534 14.8406 9.90347 14.6444C9.96223 14.5266 10.1273 14.0546 10 14.0055Z" fill="#4C4D4F"/>
<path d="M7.47491 13.4803C7.8862 13.4803 8.1506 13.1957 8.1506 12.7442V11.6352L8.61085 11.2623L8.70877 13.4705C8.70877 13.5392 8.73815 13.598 8.78711 13.6373C8.8067 13.6471 10.3441 15.0898 10.8827 16.2871C11.1471 16.8759 11.343 17.4942 11.4996 17.9849C11.6955 18.5934 11.8032 18.9271 11.9501 19.0939C12.0774 19.2313 12.2635 19.3098 12.4691 19.3098C12.8118 19.3098 13.2329 19.1037 13.3211 18.7112C13.3994 18.3971 13.1546 17.5531 12.4593 15.7081L12.4299 15.6197C12.1949 14.9916 11.4213 13.9219 11.2352 13.6569L10.9708 10.1337C10.9611 9.41726 10.2854 8.86768 9.68803 8.86768H9.65865C9.15923 8.87749 8.8067 9.07377 8.57168 9.21117L8.52271 9.24061C8.31707 9.35838 6.87757 10.4379 6.71109 10.8403C6.574 11.1642 6.66213 12.4105 6.69151 12.8031C6.7013 13.0583 6.81881 13.4803 7.47491 13.4803Z" fill="#4C4D4F"/>
<path d="M10.5889 12.2233C10.5889 12.2135 10.5889 12.2135 10.5889 12.2233C10.5791 12.2233 11.3329 12.6374 11 12.5C10.8727 12.4509 10.4064 12.1655 10.3476 12.2833C10.2987 12.4109 11.2451 12.6758 11.3626 12.7347C11.6857 12.8721 11.9599 12.9703 11.9991 12.9899C12.0872 13.0292 12.195 13.0586 12.3125 13.0586C12.5279 13.0586 12.7727 12.9506 12.949 12.5679C13.1253 12.195 12.9784 11.8318 12.5769 11.6356L11.4703 11.1056C10.5 10.5 10.5587 10.8724 10.5 11C10.4412 11.1178 10.8727 11.4411 11 11.5H10C10.2252 11.608 9.39656 11.5767 9.36719 11.6355C9.26926 11.8907 10.7162 12.2724 10.5889 12.2233Z" fill="#4C4D4F"/>
<path d="M16.7583 4H9.8056C9.66851 4 9.56079 4.10795 9.56079 4.24535V4.9814C9.56079 5.1188 9.66851 5.22675 9.8056 5.22675C9.9427 5.22675 10.0504 5.1188 10.0504 4.9814V4.4907H16.5135V19.408H10.0504V19.3099C10.0504 19.1725 9.9427 19.0645 9.8056 19.0645C9.66851 19.0645 9.56079 19.1725 9.56079 19.3099V19.6534C9.56079 19.7908 9.66851 19.8987 9.8056 19.8987H16.7583C16.8954 19.8987 17.0031 19.7908 17.0031 19.6534V4.24535C17.0031 4.10795 16.8856 4 16.7583 4Z" fill="#4C4D4F"/>
</svg>