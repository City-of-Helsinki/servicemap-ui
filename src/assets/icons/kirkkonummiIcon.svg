<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="3" y="1" width="19" height="22">
<path fill-rule="evenodd" clip-rule="evenodd" d="M21.5455 1H3L3.59091 14.2727C3.59091 16.6667 4.43939 18.7197 6.13636 20.4318C7.83334 22.1439 9.87878 23 12.2727 23C14.6667 23 16.7121 22.1439 18.4091 20.4318C20.1061 18.7197 20.9545 16.6667 20.9545 14.2727L21.5455 1ZM20.8636 1.63636L20.3182 14.2727C20.3182 16.4849 19.5303 18.3788 17.9545 19.9545C16.3788 21.5303 14.4849 22.3182 12.2727 22.3182C10.0606 22.3182 8.16667 21.5303 6.59091 19.9545C5.01514 18.3788 4.22727 16.4697 4.22727 14.2273L3.68182 1.63636H20.8636Z" fill="white"/>
</mask>
<g mask="url(#mask0)">
<rect width="24" height="24" fill="#000000"/>
</g>
<path fill-rule="evenodd" clip-rule="evenodd" d="M8.17835 11C7.20374 11 6.83304 11.7589 5.85843 11.7589C5.05382 11.7589 4.66042 11.2423 4 11.0621V11.3032C4.66042 11.484 5.05382 12 5.85843 12C6.83304 12 7.20374 11.2411 8.17835 11.2411C9.15296 11.2411 9.52366 12 10.4987 12C11.4738 12 11.844 11.2411 12.8186 11.2411C13.7932 11.2411 14.164 12 15.139 12C16.1141 12 16.4843 11.2411 17.4589 11.2411C18.4335 11.2411 18.8042 12 19.7793 12C20.296 12 20.6431 11.7866 21 11.5865V11.3454C20.644 11.5455 20.296 11.7589 19.7793 11.7589C18.8042 11.7589 18.434 11 17.4589 11C16.4839 11 16.1136 11.7589 15.139 11.7589C14.1644 11.7589 13.7937 11 12.8186 11C11.8436 11 11.4733 11.7589 10.4974 11.7589C9.52144 11.7589 9.1534 11 8.17835 11Z" fill="#000000" stroke="#000000" stroke-width="0.14"/>
<path d="M13.3786 5.25337V5.55337H13.6786H15.4149V7.64143H13.6786H13.3786V7.94143V9.71302H11.3363V7.94143V7.64143H11.0363H9.3V5.55337H11.0363H11.3363V5.25337V3.48178H13.3786V5.25337Z" stroke="#000000" stroke-width="0.6"/>
<path d="M15.0816 18.8912H9.61334L9.58227 18.8721C7.79136 17.7592 6.91487 16.4226 6.49537 15.497C6.03839 14.4886 6.00685 13.7503 6.00594 13.7192L6 13.5411L6.16908 13.5876C6.44689 13.6661 6.7195 13.7626 6.98525 13.8763L7.06705 13.9117L7.06339 14.0046C7.06339 14.0046 7.05974 14.1125 7.0707 14.2836C7.42943 14.3938 8.85978 14.8247 9.83955 14.9879L9.94511 15.0056L9.96385 15.6708C10.7543 15.7759 11.5503 15.8305 12.3475 15.8345C13.1446 15.8294 13.9407 15.7737 14.7311 15.6676L14.7498 15.0023L14.8554 14.9847C15.8351 14.8215 17.2655 14.3905 17.6242 14.2803C17.6356 14.1106 17.6315 14.0037 17.6315 14.0014L17.6279 13.9112L17.7097 13.8754C17.9756 13.7616 18.2483 13.6652 18.5263 13.5867L18.6954 13.5402L18.689 13.7183C18.689 13.7494 18.6574 14.4877 18.2 15.496C17.78 16.4217 16.9035 17.7582 15.1126 18.8712L15.0816 18.8912ZM9.68646 18.6276H15.0085C16.7171 17.5583 17.5557 16.2827 17.9587 15.3993C18.181 14.9235 18.3334 14.4171 18.4111 13.8963C18.2777 13.9386 18.0858 14.0037 17.8911 14.0823C17.8911 14.152 17.887 14.258 17.8755 14.3896L17.8678 14.4774L17.7851 14.5035C17.7682 14.5091 16.1399 15.02 15.003 15.2241L14.9838 15.8945L14.8768 15.9107C14.0386 16.0292 13.1937 16.0913 12.3475 16.0967C11.5012 16.0913 10.6563 16.0292 9.81807 15.9107L9.71114 15.8945L9.6924 15.2241C8.55452 15.0218 6.92721 14.5109 6.90939 14.5053L6.82622 14.4793L6.81799 14.3933C6.80657 14.2617 6.80337 14.1557 6.80246 14.086C6.60778 14.0056 6.41631 13.9405 6.28241 13.9C6.36013 14.4209 6.51255 14.9273 6.73482 15.4031C7.13788 16.2827 7.97827 17.5574 9.68646 18.6276Z" fill="#000000" stroke="#000000" stroke-width="0.05"/>
<path fill-rule="evenodd" clip-rule="evenodd" d="M7.04644 14.5012H6.97333C7.02908 15.3505 7.63321 16.2538 8.28578 16.8196C9.56532 17.9047 10.7078 18.2608 12.3474 18.2608C13.9871 18.2608 15.1295 17.9047 16.4095 16.8196C17.3002 16.0219 17.7096 15.4947 17.838 14.4723L17.7466 14.4668C17.7154 14.6854 17.6637 14.9006 17.5922 15.1093C16.5544 15.6048 14.9371 15.7824 14.9371 15.7824L14.949 15.8754C14.949 15.8754 16.4954 15.7062 17.5593 15.2208C17.4798 15.431 17.3472 15.6011 17.1846 15.8536C17.1242 15.8815 15.448 16.5946 12.3474 16.5946C9.24681 16.5946 7.60807 15.8619 7.54912 15.8345C7.40361 15.6134 7.28109 15.3775 7.18354 15.1306C8.24739 15.616 9.83997 15.8745 9.83997 15.8745V15.7815C9.83997 15.7815 8.18844 15.5137 7.15064 15.0181C7.02725 14.6462 7.04782 14.5002 7.04782 14.5002L7.04644 14.5012ZM7.63778 15.9684C7.99102 16.1125 9.5571 16.6876 12.3483 16.6876C15.1396 16.6876 16.7431 16.1297 17.0964 15.9875C16.9136 16.2445 16.6545 16.49 16.3556 16.7443C16.3483 16.748 14.9166 17.4728 12.3483 17.4728C9.7801 17.4728 8.34838 16.748 8.34107 16.7443C8.07508 16.5172 7.83873 16.2565 7.63778 15.9684V15.9684ZM8.59744 16.9317C8.95068 17.0874 10.1283 17.5653 12.3479 17.5653C14.5674 17.5653 15.7848 17.0809 16.1381 16.9252C14.9737 17.8359 13.8637 18.1683 12.3474 18.1683C10.8312 18.1683 9.76091 17.8424 8.59698 16.9317H8.59744Z" fill="#000000" stroke="#000000" stroke-width="0.05"/>
</svg>