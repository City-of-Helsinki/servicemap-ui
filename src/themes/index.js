import { createMuiTheme } from '@material-ui/core';

const focusIndicator = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(71, 131, 235), 0 0 0 8px rgb(255, 255, 255)',
};

const focusIndicatorDark = {
  outline: '2px solid transparent',
  boxShadow: '0 0 0 2px rgb(255, 255, 255), 0 0 0 6px rgb(0, 0, 0), 0 0 0 8px rgb(255, 255, 255)',
};

const props = {
  // Globally disable all default mui focus effects
  MuiButtonBase: {
    disableRipple: true,
  },
  MuiButton: {
    disableRipple: true,
    disableFocusRipple: true,
  },
  MuiTypography: {
    variant: 'body2',
  },
};

const overrides = theme => ({
  MuiButtonBase: {
    root: {
      // Default keyboard focus indicator for buttons
      '&$focusVisible': theme === 'dark' ? focusIndicatorDark : focusIndicator,
    },
  },
  MuiInputBase: {
    // Default keyboard focus indicator for input fields
    root: { '&$focused': theme === 'dark' ? focusIndicatorDark : focusIndicator },
  },
  MuiListItemText: {
    root: {
      marginTop: 0,
      marginBottom: 0,
    },
  },
  MuiSvgIcon: {
    root: {
      fontSize: 24,
    },
  },
  PrivateSwitchBase: {
    root: {
      padding: 12,
    },
  },
});

const breakpoints = {
  values: {
    xs: 0,
    sm: 700,
    md: 900,
    lg: 1280,
    xl: 1920,
  },
};

const spacing = 8;

const custom = {
  body2light: {
    color: 'rgba(0,0,0,0.7)',
  },
};

const zIndex = {
  behind: -1,
  forward: 50,
  sticky: 51,
  infront: 900,
};

const typography = {
  useNextVariants: true,
  fontSize: 16,
  // Use the system font instead of the default Roboto font.
  fontFamily: [
    'Lato',
  ].join(','),
  body1: {
    fontSize: '1.043rem',
    letterSpacing: '0.03125rem', // 0.5px
    lineHeight: '1.75rem',
  },
  body2: {
    fontSize: '0.913rem',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.25rem',
  },
  subtitle1: {
    fontSize: '1.03rem',
    fontWeight: 'bold',
    letterSpacing: '0.00937rem', // 0.15px
    lineHeight: '1.5rem',
  },
  caption: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.025rem', // 0.4px
    lineHeight: '1rem',
    color: 'rgba(0,0,0,0.6)',
  },
  h1: {
    fontSize: '6.259rem',
    letterSpacing: '-0.09375em', // -1.5px
  },
  h2: {
    fontSize: '3.862rem',
    fontWeight: 'bold',
    letterSpacing: '-0.03125rem', // -0.5px
  },
  h3: {
    fontSize: '3.089rem',
    fontWeight: 'bold',
    letterSpacing: 0,
  },
  h4: {
    fontSize: '2.214rem',
    letterSpacing: '0.015625rem', // 0.25px
  },
  h5: {
    fontSize: '1.564rem',
    letterSpacing: 0,
  },
  h6: {
    fontSize: '1.288rem',
    fontWeight: 'bold',
    letterSpacing: '0.015625rem', // 0.25px
    lineHeight: '1.5rem',
  },
  button: {
    fontSize: '0.90125rem', // 14.42px
    fontWeight: 'bold',
    letterSpacing: '0.078125rem', // 1.25px
    lineHeight: '1rem', // 16px
    color: '#ffffff',
  },
  overline: {
    fontSize: '0.7725rem', // 12.36px
    fontWeight: 'bold',
    letterSpacing: '0.125rem', // 2px
    lineHeight: '1rem',
  },
};

// Color palette for normal theme
export const paletteDefault = {
  primary: {
    main: 'rgb(25, 100, 230)',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#1d39ad',
  },
  background: {
    main: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
    plain: 'rgb(20, 108, 232)',
    front: 'linear-gradient(340.58deg, rgb(11, 123, 237) 0%, rgb(20, 108, 232) 67.04%, rgb(25, 100, 230) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: 'rgb(25, 100, 230)',
    alpha: 'rgba(25,100,230,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  focusBorder: {
    main: 'rgb(71, 131, 235)',
  },
  warning: {
    main: '#AF0021',
  },
};

// Color palette for dark theme
export const paletteDark = {
  primary: {
    main: '#353638',
    highContrast: '#fff',
  },
  secondary: {
    main: 'rgb(43, 47, 57)',
    hover: '#5b5d61',
  },
  background: {
    main: '#4A4A4C',
    plain: '#4A4A4C',
    front: 'linear-gradient(326.21deg, rgba(0, 0, 0, 0.79) 0%, rgba(71, 71, 71, 0.79) 100%)',
  },
  white: {
    light: '#f2f2f2',
    main: '#FFFFFF',
    dark: '#949494',
    contrastText: '#000',
  },
  detail: {
    main: '#585B63',
    alpha: 'rgba(88,91,99,0.5)',
  },
  disabled: {
    main: 'rgb(239, 239, 239)',
    strong: '#898989',
  },
  warning: {
    main: '#AF0021',
  },
  focusBorder: {
    main: '#000',
  },
};

// Themes
const SMTheme = createMuiTheme({
  props,
  overrides: overrides('default'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDefault,
  zIndex,
});

const SMThemeDark = createMuiTheme({
  props,
  overrides: overrides('dark'),
  breakpoints,
  typography,
  spacing,
  custom,
  palette: paletteDark,
  zIndex,
});


export default { SMTheme, SMThemeDark };
