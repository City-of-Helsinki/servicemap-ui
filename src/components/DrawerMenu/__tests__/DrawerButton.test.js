// Link.react.test.js
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import { MuiThemeProvider, Typography } from '@material-ui/core';
import { Search } from '@material-ui/icons';
import DawerButton from '../DrawerButton';
import themes from '../../../themes';

// Generic required props for ServiceMapButton
const buttonMockProps = {
  active: false,
  disabled: false,
  disableRipple: true,
  icon: <Search />,
  isOpen: false,
  text: 'Drawer button text',
  onClick: () => {},
  subText: 'Drawer button subtext',
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <MuiThemeProvider theme={themes.SMTheme}>
    {children}
  </MuiThemeProvider>
);

describe('<DrawerButton />', () => {
  // let render;
  let mount;

  beforeEach(() => {
    // render = createRender({ wrappingComponent: Providers });
    mount = createMount({ wrappingComponent: Providers });
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('should work', () => {
    const component = mount(<DawerButton {...buttonMockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('does show text', () => {
    const component = mount(<DawerButton {...buttonMockProps} />);
    expect(component.find(Typography).text()).toEqual(buttonMockProps.text);
  });

  it('simulates click event', () => {
    const mockCallBack = jest.fn();
    const component = mount(
      <DawerButton
        {...buttonMockProps}
        onClick={mockCallBack}
      />,
    );
    component.simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('does render default accessibility attributes correctly', () => {
    const component = mount(
      <DawerButton
        {...buttonMockProps}
      />,
    );
    const buttonBase = component.find('ForwardRef(ButtonBase)');
    // Expect role to be link
    expect(buttonBase.props().role).toEqual('link');
    // Expect aria-label to be same as given text
    expect(buttonBase.props()['aria-label']).toEqual(buttonMockProps.text);
    // Expect visible text to be hidden from screen readers
    expect(component.find('span').props()['aria-hidden']).toEqual(true);
  });

  it('does hide correctly when menu not open', () => {
    const component = mount(
      <DawerButton
        {...buttonMockProps}
        isOpen={false}
      />,
    );

    const buttonBase = component.find('ForwardRef(ButtonBase)');
    // Expect element to be aria-hidden when menu not opened
    expect(buttonBase.props()['aria-hidden']).toEqual(true);
    // Expect tabIndex to be -1 when menu not opened
    expect(buttonBase.props().tabIndex).toEqual(-1);
  });

  it('does show correctly when menu is opened', () => {
    const component = mount(
      <DawerButton
        {...buttonMockProps}
        isOpen
      />,
    );

    const buttonBase = component.find('ForwardRef(ButtonBase)');
    // Expect element to not be hidden when menu open
    expect(buttonBase.props()['aria-hidden']).toEqual(false);
    // Expect element to be in tab order when menu is open
    expect(buttonBase.props().tabIndex).toEqual(0);
  });

  it('does render aria-label correctly when disabled', () => {
    const component = mount(
      <DawerButton
        {...buttonMockProps}
        disabled
      />,
    );
    const buttonBase = component.find('ForwardRef(ButtonBase)');
    // Expect aria-label to be same as given text
    expect(buttonBase.props()['aria-label']).toEqual(`${buttonMockProps.text} ${buttonMockProps.subText}`);
  });
});
