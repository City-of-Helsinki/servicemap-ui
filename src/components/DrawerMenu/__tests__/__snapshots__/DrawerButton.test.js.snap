// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerButton /> should work 1`] = `
<WithStyles(ButtonBase)
  aria-hidden={true}
  aria-label="Drawer button text"
  className="DrawerButton-drawerButton-3 "
  disableRipple={true}
  focusVisibleClassName="DrawerButton-itemFocus-9"
  key="Drawer button text"
  onClick={[Function]}
  role="link"
  tabIndex={-1}
>
  <div
    className="DrawerButton-drawerIcon-6  "
  >
    <pure(SearchIcon) />
  </div>
  <span
    aria-hidden={true}
  >
    <WithStyles(Typography)
      className="DrawerButton-drawerButtonText-5 "
      variant="body1"
    >
      Drawer button text
    </WithStyles(Typography)>
  </span>
</WithStyles(ButtonBase)>
`;
